.github/workflows/critical-rules-test.yml

```diff
@@ -0,0 +1,39 @@
+#############################
+## GitHub Actions CI Tests ##
+#############################
+#
+# This lints all our dialect fixtures to check rules can handle a variety
+# of SQL and don't error out badly.
+#
+# It's run as a separate job as takes longer than the CI jobs and allows
+# them to be rerun separately if GitHub Actions or Coverage is experiencing
+# issues.
+#
+name: Critical CI Tests
+on:
+  workflow_dispatch:
+  schedule:
+    # 2am each night
+    - cron: '00 2 * * *'
```

Comment:
Since it's running on a schedule, how will we be alerted if it fails? Email?

Reply:
The GitHub altering on scheduled runs is a bit lacking to be honest. Long been complained about: https://github.community/t/no-email-notification-on-schedule-cron-job-failure/119638/5

I think only the last person to alter it (me!) gets a notification. Which sucks to be honest.

Now to be honest this cron job isn’t really needed. I like it in case there is no activity on repo for a while (not something we have a problem with!!) so can see when something broke if, for example, dependencies update.